processor/otel/metrics.go

```diff
@@ -89,6 +90,172 @@ func (c *Consumer) convertResourceMetrics(resourceMetrics pdata.ResourceMetrics,
+}
+
+// accumulate processes m, translating to and accumulating equivalent Elastic APM metrics in b.
+func (b *apmMetricsBuilder) accumulate(m pdata.Metric) {
+
+	switch m.DataType() {
+	case pdata.MetricDataTypeSum:
+		dpsCounter := m.Sum().DataPoints()
+		for i := 0; i < dpsCounter.Len(); i++ {
+			dp := dpsCounter.At(i)
+			if sample, ok := numberSample(dp, model.MetricTypeCounter); ok {
+				switch m.Name() {
+				case "system.memory.usage":
+					if memoryState, exists := dp.Attributes().Get("state"); exists {
+						switch memoryState.StringVal() {
+						case "used":
+							b.usedMemoryBytes = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+						case "free":
+							b.freeMemoryBytes = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+						}
+					}
+				case "runtime.jvm.gc.collection", "runtime.jvm.gc.time":
+					if gcName, exists := dp.Attributes().Get("gc"); exists {
+						b.jvmGCTime[gcName.StringVal()] = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+					}
+				case "runtime.jvm.gc.count":
+					if gcName, exists := dp.Attributes().Get("gc"); exists {
+						b.jvmGCCount[gcName.StringVal()] = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+					}
+				}
+			}
+		}
+	case pdata.MetricDataTypeGauge:
+		// Gauge metrics accumulation
+		dpsGauge := m.Gauge().DataPoints()
+		for i := 0; i < dpsGauge.Len(); i++ {
+			dp := dpsGauge.At(i)
+			if sample, ok := numberSample(dp, model.MetricTypeGauge); ok {
+				switch m.Name() {
+				case "system.cpu.utilization":
+					if cpuState, exists := dp.Attributes().Get("state"); exists {
+						if cpuState.StringVal() != "idle" {
+							if sample.Value > 1 {
+								b.nonIdleCPUUtilizationSum.value += 1
+							} else {
+								b.nonIdleCPUUtilizationSum.value += sample.Value
+							}
+							b.nonIdleCPUUtilizationSum.timestamp = dp.Timestamp().AsTime()
+						}
+					}
+					if cpuIdStr, exists := dp.Attributes().Get("cpu"); exists {
+						cpuId, _ := strconv.Atoi(cpuIdStr.StringVal())
+						if cpuId+1 > b.cpuCount {
+							b.cpuCount = cpuId + 1
+						}
+					}
+				case "runtime.jvm.memory.area":
+					var key jvmMemoryKey
+					dp.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+						switch k {
+						case "area":
+							key.area = v.AsString()
+						case "type":
+							key.type_ = v.AsString()
+						}
+						return true
+					})
+					if key.area != "" && key.type_ != "" {
+						b.jvmMemory[key] = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+					}
+				}
+			}
+		}
+	}
+}
+
+// emit upserts Elastic APM metrics into ms from information accumulated in b.
+func (b *apmMetricsBuilder) emit(ms metricsets) {
+	// system.memory.actual.free
+	// Direct translation of system.memory.usage (state = free)
+	if b.freeMemoryBytes.value > 0 {
+		ms.upsertOne(
+			b.freeMemoryBytes.timestamp, "system.memory.actual.free", pdata.NewAttributeMap(),
+			model.MetricsetSample{Type: model.MetricTypeCounter, Value: b.freeMemoryBytes.value},
```

Comment:
This is a gauge metric (point in time), no? Counter means that it's monotonically increasing, e.g. bytes of memory allocated since process creation.

We don't currently do anything with counter/gauge, so I'd suggest we just omit Type for the moment.

Reply:
Good point ! Type omitted in 8d501af